eclipse/test/junit/saros/editor/internal/EditorAPITextPositionTest.java

```diff
@@ -0,0 +1,335 @@
+package saros.editor.internal;
+
+import static org.junit.Assert.assertEquals;
+
+import org.easymock.EasyMock;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITextSelection;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.texteditor.IDocumentProvider;
+import org.eclipse.ui.texteditor.ITextEditor;
+import org.junit.Test;
+import saros.editor.text.TextPosition;
+import saros.editor.text.TextSelection;
+
+/** Tests the position calculation logic of the editor api. */
+public class EditorAPITextPositionTest {
+
+  /**
+   * Build an IDocument mock that can be used to call {@link EditorAPI#calculateSelection(IDocument,
+   * int, int)}.
+   *
+   * <p>The needed line offset lookup answers must be set using {@link
+   * #withLineOffsetLookupAnswer(int, int)} before calling {@link #build()}.
+   *
+   * <p>The needed line number lookup answers must be set using {@link
+   * #withLineNumberLookupAnswer(int, int)} before calling {@link #build()}.
+   */
+  private static class IDocumentBuilder {
+    private IDocument document;
+
+    private IDocumentBuilder() {
+      document = EasyMock.createNiceMock(IDocument.class);
+    }
+
+    /**
+     * Sets the passed values for the line offset lookup.
+     *
+     * <p>This must be called for all needed lookup values before calling {@link #build()}.
+     *
+     * @param lineNumberInput the line number whose offset is looked up
+     * @param lineOffsetAnswer the offset to return as the answer for the lookup
+     * @return this builder
+     */
+    private IDocumentBuilder withLineOffsetLookupAnswer(int lineNumberInput, int lineOffsetAnswer) {
+      try {
+        EasyMock.expect(document.getLineOffset(lineNumberInput)).andReturn(lineOffsetAnswer);
+
+      } catch (BadLocationException ignored) {
+      }
```

Comment:
please make the eclipse problem marker happy and add NOP comments :)

Reply:
This is handled with the name "ignored" for the IntelliJ analyze. But I am more than happy to also add a comment.